lang/Makefile.common

```diff
@@ -0,0 +1,68 @@
+#
+# Generic makefile to build and test a parser for a specific language.
+#
+
+PROJECT_ROOT = $(shell git rev-parse --show-toplevel)
```

Comment:
nice! I was always manually doing ../../../

Reply:
The only thing is it depends on git and being inside of a valid git repo, i.e. it doesn't work if you distribute a source tarball without the `.git`.